### DISCUSSION TITLE
DAI 2024-2025 - My picocli project - Mathieu Emery

### DISCUSSION THREAD

**Discussion by mathieuemery** [heig-vd-dai-course/heig-vd-dai-course#233] [Show and tell]
https://github.com/mathieuemery/DAI-04/

**Comment by ludelafo** [comment]
Good usage of ChatGPT for the README! And thank you for mentioning it.

I see that your `pom.xml` file contains the following lines:

```xml
    <properties>
        <!-- Omitted for brevity -->
    </properties>
```

I guess it's a direct copy/paste from the course material, but you must leave the initial values of these properties. They define the Java version as well as the encoding character (more on this in a later chapter) used to compile the project.

In the same vein, your README states that Java 11 or later must be used to compile the project. Just to be sure, have you indeed installed Java 21 with SDKMAN! on your computer? Maybe it's a typo/mistake in the README but I would update it with the right information just to be sure everyone can run your project on their computer.

I see that the gitignore file (`.gitignore`) contains other files for other IDEs (Visual Studio Code for example). While totally acceptable, I tend to avoid to keep "unused" rules in my gitignore files to avoid noise and keep it neat and tidy. The course material asks to replace all content with the one provided in the course material to avoid unnecessary noise.

You can always compare your solutions with the official ones available in the following repository: <https://github.com/heig-vd-dai-course/heig-vd-dai-course-solutions>.

Feel free to let us know if everything was clear. :)

**Comment by hadrylouis** [comment]
Nice work, well done.

In addition to @ludelafo comments, files with the *.java extension seem to have bad indentation. Would it be possible to improve this ?

**Comment by mathieuemery** [comment]
Hello everyone,

Thank you for the feedbacks, I did the corrections.

I indeed installed java 21 so there was a typo in the Readme.md.

**Reply by ludelafo** [reply]
Perfect! Good luck with the army. :)
